File: sdk/java/src/main/java/domain/DataSource.java
package domain;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class DataSource {

    private Map data;

    /**
     * A DataSource is a remote source for Assets that can be
     * iterated by the Analysis framework and imported
     * in a single import Job.
     *
     * @param data Contains DataSource Attributes
     */
    public DataSource(Map data) {
        this.data = data;
    }

    /**
     *
     * @return The id of the DataSource
     */
    public String getId() {
        return (String) this.data.get("id");
    }

    /**
     *
     * @return The name of the DataSource
     */
    public String getName() {
        return (String) this.data.get("name");
    }

    /**
     *
     * @return The URI of the DataSource
     */
    public String getUri() {
        return (String) this.data.get("uri");
    }

    /**
     *
     * @return The file type filter for the DataSource
     */
    public List<String> getFileTypes() {
        return (List<String>) Optional.ofNullable(this.data.get("file_types")).orElse(new ArrayList());
    }

    /**
     *
     * @return The type of analysis done to the DataSource
     */

    public List getAnalysis() {
        return (List) Optional.ofNullable(this.data.get("analysis")).orElse(new ArrayList());

    }
}

File: client/python/pylib/zmlp/datasource.py
class DataSource(object):
    """
    A DataSource is a remote source for Assets that can be
    iterated by the Analysis framework and imported
    in a single import Job.
    """

    def __init__(self, data):
        self._data = data

    @property
    def id(self):
        """The id of the DataSource"""
        return self._data['id']

    @property
    def name(self):
        """The name of the DataSource"""
        return self._data['name']

    @property
    def uri(self):
        """The URI of the DataSource"""
        return self._data['uri']

    @property
    def file_types(self):
        """The file type filter for the DataSource"""
        return self._data.get('file_types', [])

    @property
    def analysis(self):
        """The type of analysis done to the DataSource"""
        return self._data.get('analysis', [])
